sympy/assumptions/tests/test_rel_queries.py

```diff
@@ -151,12 +158,12 @@ def test_failing_number_line_properties():
 
 def test_equality():
     # test symetry and reflexivity
-    assert ask(Q.eq(x, x)) is True
-    assert ask(Q.eq(y, x), Q.eq(x, y)) is True
-    assert ask(Q.eq(y, x), ~Q.eq(z, z) | Q.eq(x, y)) is True
+    assert ask(Q.eq(x, x), Q.real(x)) is True
+    assert ask(Q.eq(y, x), Q.eq(x, y) & Q.real(x) & Q.real(y)) is True
+    assert ask(Q.eq(y, x), (~Q.eq(z, z) | Q.eq(x, y)) & Q.real(x) & Q.real(y) & Q.real(z)) is True
 
     # test transitivity
-    assert ask(Q.eq(x,z), Q.eq(x,y) & Q.eq(y,z)) is True
+    assert ask(Q.eq(x,z), Q.eq(x,y) & Q.eq(y,z) & Q.real(x) & Q.real(y) & Q.real(z)) is True
```

Comment:
I guess we need a different (simpler) solver for cases like this that do not involve inequalities.

Reply:
This can be fixed by checking to make sure all expressions that are part of inequalities (but not those part of equalities or unequalities) are real instead of checking to make sure all expressions are real.